app/controllers/sessions_controller.rb

```diff
@@ -157,6 +160,16 @@ def create
           end
 
           redirect_to root_path, alert: I18n.t(:'controllers.lost_user')
+        elsif unverified
+            Rails.logger.warn do
+              oauth = request.env['omniauth.auth']
+              errors = @handler_result.errors.inspect
+
+              "[SessionsCreate] Unverified account: " +
+              "OAuth data: #{oauth}; status: #{status}; errors: #{errors}"
+            end
+
+            redirect_to root_path, alert: I18n.t(:'controllers.lost_user')
```

Comment:
Now that we have two things we are looking for, can we pull this error code into a helper method?

```ruby
def log_and_send_to_root(dev_message, user_alert)
  ...
end
```

```ruby
error_codes = errors.map(&:code)

if error_codes.include?(:unknown_callback_state) || error_codes.include?(:invalid_omniauth_data)
  log_and_send_to_root("Lost user on failure", I18n.t(:'controllers.lost_user'))
elsif (error_codes.include?(:not_verified)
  log_and_send_to_root("Unverified account", I18n.t(:'controllers.lost_user')) # need a different message here.
else
 ...
end
```

Reply:
Need a different message displayed to the user, because I think we know more about how this happens.  That verbiage is in `config/locales/en.yml`.

Reply:
Is this test roughly what we're looking for? I think it's what's described on the card.
https://github.com/openstax/accounts/blob/fix-email-validation-message-appears-twice/spec/features/user_signs_up_spec.rb#L697-L719

Reply:
Looks similar except user wouldn't complete verification and may or may not need the 2nd browser.